lib/src/zenohUTransport.cpp

```diff
@@ -459,16 +467,15 @@ void ZenohUTransport::QueryHandler(const z_query_t *query, void *arg) {
 
     auto uuidStr = UuidSerializer::serializeToString(attributes.id());
 
-    instance->queryMap_[uuidStr] = z_query_clone(query);
-
     if (UMessageType::UMESSAGE_TYPE_REQUEST != attributes.type()) {
        spdlog::error("Wrong message type = {}", static_cast<int>(attributes.type()));
        return;
     }
 
-    std::unique_lock<std::mutex> lock(instance->queryMapMutex_);
-    instance->queryMap_[uuidStr] = z_query_clone(query);
-    lock.unlock();
+    {
```

Comment:
Why is this inside of the brackets?

Reply:
To limit the scope of the lock. It also creates an explicit visual grouping so that the locked context can be seen easily by anyone reading the code.